app/assets/javascripts/shared/dradis_datatable.js

```diff
@@ -2,35 +2,129 @@ class DradisDatatable {
   constructor(tableElement) {
     this.$table = $(tableElement);
     this.dataTable = null;
+    this.tableHeaders = Array.from(this.$table[0].querySelectorAll('thead th, thead td'));
     this.init();
   }
 
   init() {
-    var dataTable = this.$table.DataTable({
+    var that = this;
+
+    // Remove dropdown option for <th> columns that has data-colvis="false" in colvis button
+    var colvisColumnIndexes = [];
+    this.tableHeaders.forEach(function(column, index) {
+      if(column.dataset.colvis != 'false') {
+        colvisColumnIndexes.push(index);
+      }
+    });
+
+    // Assign the instantiated DataTable as a DradisDatatable property
+    this.dataTable = this.$table.DataTable({
       autoWidth: false,
-      dom: "<'row'<'col-lg-6'><'col-lg-6'f>>" +
-      "<'row'<'col-lg-12'tr>>" +
-      "<'dataTables_footer_content'ip>",
+      buttons: {
+        dom: {
+          button: {
+            tag: 'button',
+            className: 'btn'
+          }
+        },
+        buttons: [
+          {
+            available: function(){
+              return that.$table.find('td.select-checkbox').length;
+            },
+            attr: {
+              id: 'select-all'
+            },
+            name: 'selectAll',
+            text: '<label for="select-all-checkbox" class="sr-only">Select all"</label><input type="checkbox" id="select-all-checkbox" />',
+            titleAttr: 'Select all'
+          },
+          {
+            extend: 'colvis',
+            text: '<i class="fa fa-columns mr-1"></i><i class="fa fa-caret-down"></i>',
+            titleAttr: 'Choose columns to show',
+            className: 'btn',
+            columns: colvisColumnIndexes
+          }
+        ]
+      },
+      dom: "<'row'<'col-lg-6'B><'col-lg-6'f>>" +
+        "<'row'<'col-lg-12'tr>>" +
+        "<'dataTables_footer_content'ip>",
       initComplete: function (settings) {
         settings.oInstance.wrap("<div class='table-wrapper'></div>");
       },
-      pageLength: 25
+      lengthChange: false,
+      pageLength: 25,
+      select: {
+        selector: 'td:first-child',
```

Comment:
Shouldn't this be `td.select-checkbox`?

Reply:
Both works, but I like the `select-checkbox` more for the specificity.